elements/lisk-p2p/src/p2p.ts

```diff
@@ -795,13 +820,20 @@ export class P2P extends EventEmitter {
 				...peer.sharedState,
 			}));
 
-		request.end({
-			success: true,
-			peers:
-				getByteSize(sanitizedPeerInfoList) < wsMaxPayload
-					? sanitizedPeerInfoList
-					: sanitizedPeerInfoList.slice(0, safeMaxPeerInfoLength),
-		});
+		const peersList =
+			getByteSize(sanitizedPeerInfoList) < wsMaxPayload
+				? sanitizedPeerInfoList
+				: sanitizedPeerInfoList.slice(0, safeMaxPeerInfoLength);
+		const encodedPeersList = peersList.map(peer =>
+			codec.encode(peerInfoSchema, peer),
+		);
+
+		request.end(
+			codec.encode(peersListResponseSchema, {
```

Comment:
As we discussed, the response can be JSON format, so no need to encode the out most structure.